packages/twoside/init.lua

```diff
@@ -15,10 +15,10 @@ local mirrorMaster = function (_, existing, new)
   for name, frame in pairs(SILE.scratch.masters[existing].frames) do
     local newframe = pl.tablex.deepcopy(frame)
     if frame:isAbsoluteConstraint("right") then
-      newframe.constraints.left = "100%pw-("..frame.constraints.right..")"
+      newframe.constraints.left = "2 * left(page) + 100%pw-("..frame.constraints.right..")"
```

Comment:
Good catch here. I've been holding off merging this because it is a breaking change that I haven't had time to regression test and figure out how to transition people onto yet, but this fix should go in a patch release even without a major version bump. I'm going to stuff it in #1562.

Reply:
>  it is a breaking change that I haven't had time to regression test and figure out how to transition people onto yet

It's indeed slightly breaking (people who did neither use cropmarks nor redefined the papersize "in the wild" wouldn't even notice it - That's possible the general case). There's no urge on this PR, as long as we know it exists and might be a way to address full-bleed printing.

Actually I was kind of tempted to split the `\background` fixes and improvements (dependency on color, ability to use an image) and the crop header fix (date issue) separately, would we want them earlier (true, it wouldn't account for page bleeds in print, but that's already the case for a colored background). Not sure, but this could have an interest for people just producing a PDF and not requiring page bleeds. It's a bit of extra work, but if you'd consider the approach more reasonable, I could do such a split (and thenlater rebase this PR over it).

Reply:
I had to pull this fix from #1562 because it breaks the default book class. It only works correctly in the context of all frames being relative as you've done in this PR ... which makes it a breaking change that shouldn't go in a patch release.

I'd wait to split this up, I'm going to take a stab at it along with a compatibility shim, but I'm going to have to ship my next book tomorrow with just a hacked up mirror function.

Reply:
I have more doubts than ever, having dabbled into dark horizons: the typesetter, pagebuilder, frame and insertion mechanisms are very entangled and touchy, with plenty of assumptions.

For instance, the X,Y cursors in a frame are not initialized when the frames are (re-)solved. Playing with the (undocumented) `steal-position` insertion for top-growing insertions (for #1559) causes much havoc :)

In other terms, I start thinking that this PR comes too early, in a way, and might be premature.

As noted, the `\background` fixes and improvements (dependency on color, ability to use an image) and the crop header fix (date issue) could still make sense outside of this picture.

Reply:
Nah. Some ideas here might be reusable, but definitively, a different approach is needed.
Possibly the whole bleed question is an outputter thing only...